### PR TITLE
[15.0][IMP] dms_field: Add support to drag & drop

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT48179

ping @victoralmau @chienandalu 

With this changes we add the drag and drop option for dms fields. On next gif you can see an example:
Ejemplo dms_field

### CODE REVIEW

File: dms_field/static/src/js/views/many_drop_target.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/* global base64js*/
+/* Copyright 2020 Creu Blanca
+ * Copyright 2021 Tecnativa - Alexandre D. Díaz
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define("dms_field.DragDrop", function (require) {
```
Comment (chienandalu) [inline dms_field/static/src/js/views/many_drop_target.js]: Is this copied from somewhere else?

**Comment by CarlosRoca13** [comment]
Changed

**Review by victoralmau — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Does this need to depend on `web_drop_target`?

**Comment by CarlosRoca13** [comment]
dms is depending yet on that module

**Review by legalsylvain — APPROVED** [review]
Hi @CarlosRoca13. Nice work !

tested on runboat. works like a charm.

**Comment by pedrobaeza** [comment]
Isn't possible to preserve previous folding state in the tree when dragging the file? Maybe adding the element by hand with jstree API instead of refreshing whole one.

**Comment by CarlosRoca13** [comment]
What do you mean? When I drag the file I don't see the directories folding...
keepling_state

**Comment by pedrobaeza** [comment]
The problem is reproducible when you drop on a folder with subfolders that are folded:

Peek 11-03-2024 08-36

**Comment by CarlosRoca13** [comment]
@pedrobaeza Solved and improved the code to avoid full recharging when adding new directory or new field using the wizard :smile:

**Comment by pedrobaeza** [comment]
Still incorrect. Now it folds all the subfolders:

Peek 11-03-2024 13-01

**Comment by CarlosRoca13** [comment]
It should be working correctly right now :)

**Review by pedrobaeza — APPROVED** [review]
Great, thanks.

/ocabot merge minor

**Pull request merged**